scenedetect/detectors/adaptive_content_detector.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+#
+#         PySceneDetect: Python-Based Video Scene Detector
+#   ---------------------------------------------------------------
+#     [  Site: http://www.bcastell.com/projects/PySceneDetect/   ]
+#     [  Github: https://github.com/Breakthrough/PySceneDetect/  ]
+#     [  Documentation: http://pyscenedetect.readthedocs.org/    ]
+#
+# Copyright (C) 2014-2019 Brandon Castellano <http://www.bcastell.com>.
+#
+# PySceneDetect is licensed under the BSD 3-Clause License; see the included
+# LICENSE file, or visit one of the following pages for details:
+#  - https://github.com/Breakthrough/PySceneDetect/
+#  - http://www.bcastell.com/projects/PySceneDetect/
+#
+# This software uses Numpy, OpenCV, click, tqdm, simpletable, and pytest.
+# See the included LICENSE files or one of the above URLs for more information.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+""" Module: ``scenedetect.detectors.content_detector``
+
+This module implements the :py:class:`ContentDetector`, which compares the
+difference in content between adjacent frames against a set threshold/score,
+which if exceeded, triggers a scene cut.
+
+This detector is available from the command-line interface by using the
+`detect-content` command.
+"""
+
+# Third-Party Library Imports
+import numpy
+import cv2
+
+# PySceneDetect Library Imports
+from scenedetect.scene_detector import SceneDetector
+
+
+class AdaptiveContentDetector(SceneDetector):
+    """Detects cuts using HSV changes similar to ContentDetector, but with a
+    rolling average that can help mitigate false detections in situations such
+    as camera moves.
+    """
+
+    def __init__(self, video_manager=None, adaptive_threshold=3.0, min_scene_len=15, min_delta_hsv=5.0, window_width=2):
+        super(AdaptiveContentDetector, self).__init__()
```

Comment:
Technically this could inherit from ContentDetector to reduce a lot of code duplication - thoughts?  I don't mind taking that one on, so not a huge deal - it can always be done after merging anyways if that's the case.

Reply:
I agree about chatting and happy to do so (my email should be on my gihub profile, or `doctor_walter#7001` on discord). I think the idea of inheriting from `ContentDetector` is a good idea. I don't currently have a dev environment set up, so I can work on getting that set up to poke around with it.

Reply:
The latest commit added inheritance, allowing for the `process_frame` function to be completely inherited. The trick to it is to initialize the `ContentDetector` with an impossibly high threshold so that it doesn't trigger any cuts. This leaves the cut list empty until the `post_process` function analyzes the metrics and writes the cuts.

Reply:
Gotcha - I think the base class method could be called instead, and have the return value discarded (rather than overriding the threshold).  Thoughts?